R/execution.r

```diff
@@ -129,6 +129,27 @@ execute = function(request) {
     # shade base::quit
     assign('quit', quit, envir = .GlobalEnv)
     assign('q',    quit, envir = .GlobalEnv)
+
+    # This is needed to get htmlwidgets displayed on print
+    print.htmlwidget <- function(x, ...){
+        htmlfile <- tempfile(fileext = ".html")
+
+        # save the widget to HTML
+        htmlwidgets::saveWidget(
+            widget = x,
+            file = htmlfile,
+            selfcontained = TRUE
+        )
+
+        html_string <- readChar(htmlfile, file.info(htmlfile)$size)
+        content <- namedlist()
+        metadata <- namedlist()
+        content[['text/html']] <- html_string
+        metadata[['text/html']] <- list(isolated = TRUE)
+        publish_mimebundle(content, metadata)
+    }
+
+    assign('print.htmlwidget', print.htmlwidget, envir = .GlobalEnv)
```

Comment:
I suspect that we end up with a few more such functions (quit is already there; cran chooser, debug stuff, password input...). Is there a way to refactor this into a new file? Probably be easiest would be if there is a way to add a new environment so that it is available both in the kernel and for the user and the executer only adds it?